### DISCUSSION TITLE
Why is Http::fake() not working?

### DISCUSSION THREAD

**Discussion by pikant** [saloonphp/saloon#388] [General]
Hello!

In the [documentation](https://docs.saloon.dev/installable-plugins/laravel-integration#laravel-http-client-sender) it says the following:

> The Laravel HTTP sender plugin does not support the Http::fake() method.

Why is that actually?

In my opinion, one of Saloon's greatest assets is that it uses Laravel's HTTP client under the hood if you configure it accordingly. And `Http::fake()` is one of the largest assets of the Laravel HTTP client.

Are there any plans to add Http::fake() functionality when using the Laravel HTTP sender?

**Comment by Sammyjo20** [comment]
Hey @pikant I hope you are well.

It's a good question. The implementation of the HTTP client isn't as simple as it may appear to be, and therefor some features like `Http::fake()` are not available. That being said, if you are using Saloon I wouldn't recommend using `Http::fake()` because you can use `Saloon::fake()` instead. Saloon has the same if not more testing utilities than Laravel's built in helpers including the ability to record requests.

There are no plans to add the functionality, in fact there are plans to slowly sunset the HTTP client library as I feel it is adding additional overhead when it's not needed.

**Reply by pikant** [reply]
Hey @Sammyjo20,

I hope you are doing great too! Thanks a lot for your quick answer and honest opinion.

The reason why we would use the Laravel HTTP sender is that we love Saloon's way of structuring things, but we want also to keep things as close to the Laravel standard as possible - technically speaking.

> There are no plans to add the functionality, in fact, there are plans to slowly sunset the HTTP client library as I feel it is adding additional overhead when it's not needed.

Keeping in mind, that Saloon is a general PHP library, this does make sense. In this case, however, we will look for an alternative that is based on the Laravel Http client and allows Http::fake(), or implement something of our own :)

Many thanks!
